src/common/types/vector.cpp

```diff
@@ -1358,6 +1358,18 @@ void ListVector::Append(Vector &target, const Vector &source, const SelectionVec
 	auto &target_buffer = (VectorListBuffer &)*target.auxiliary;
 	target_buffer.Append(source, sel, source_size, source_offset);
 }
+ValidityMask &ListVector::GetValidityMask(Vector &vector) {
+	D_ASSERT(vector.GetType().id() == LogicalTypeId::LIST);
+	if (vector.GetVectorType() == VectorType::DICTIONARY_VECTOR) {
+		auto &child = DictionaryVector::Child(vector);
+		return FlatVector::Validity(ListVector::GetEntry(child));
```

Comment:
It seems dangerous to return a validity mask of a child of a dictionary vector as-is: without the dictionary you do not know how to interpret the validity mask. Perhaps use `Orrify` instead?